cpp/daal/src/algorithms/kmeans/oneapi/cl_kernels/kmeans_cl_kernels.cl

```diff
@@ -156,7 +157,8 @@ DECLARE_SOURCE(
             }
         }
         int curInd = 1;
-        if (minIdx < numgrp - 1 || local_id < rem) curInd = (distances[global_id + N * (local_id + minIdx * size)] > minVal) ? 1 : -local_id;
+        if (minIdx < numgrp - 1 || local_id < rem)
+            curInd = (distances[global_id + N * (local_id + minIdx * size)] + centroidsSq[local_id + minIdx * size] > minVal) ? 1 : -local_id;
         int ind = minIdx * size - sub_group_reduce_min(curInd);
         if (local_id == 0)
```

Comment:
line 166: we assigning not a distance to `mindistances`, but distance + centroidSq. Where is the mistake?

Reply:
It is a change introduced. Distances are computed without initialization of matrix with centroidsSq